File: frontend/src/app/models/match/match-item.response.ts
import { TeamRef } from "../common/team-ref.model";
import { UserRef } from "../common/user-ref.model";

export interface MatchItemResponse {
    id: number;
    name: string | 'Match';
    status: 'SCHEDULED' | 'NOT_SCHEDULED' | 'COMPLETED';
    teamHome: TeamRef | null;
    teamAway: TeamRef | null;
    referee: UserRef | null;
}

File: src/main/java/com/tournament/app/footycup/backend/dto/match/MatchItemResponse.java
package com.tournament.app.footycup.backend.dto.match;

import com.tournament.app.footycup.backend.dto.common.TeamRef;
import com.tournament.app.footycup.backend.dto.common.UserRef;

public record MatchItemResponse(
    Long id,
    String name,
    String status,
    TeamRef teamHome,
    TeamRef teamAway,
    UserRef referee
    ) {
}
